crates/ffi/src/lib.rs

```diff
@@ -238,11 +259,8 @@ macro_rules! generate_lib {
 
 /// # Safety
 /// `ArrowArray` and `ArrowSchema` must be valid
-pub unsafe fn import_array(
-    array: FFI_ArrowArray,
-    schema: &FFI_ArrowSchema,
-) -> datafusion::error::Result<ArrayRef> {
-    let data = datafusion::arrow::ffi::from_ffi(array, schema).unwrap();
-    let arr = datafusion::arrow::array::make_array(data);
+pub unsafe fn import_array(array: FFI_ArrowArray, schema: &FFI_ArrowSchema) -> Result<ArrayRef> {
+    let data = arrow::ffi::from_ffi(array, schema).unwrap();
+    let arr = arrow::array::make_array(data);
     Ok(arr)
 }
```

Comment:
seems like we should either unwrap/panic or return a result, but the current state of "panic and never return an error result" is either cruft or a mistake.